dev/NumberBox/NumberBoxParser.cpp

```diff
@@ -0,0 +1,295 @@
+﻿#include "pch.h"
+#include "NumberBoxParser.h"
+
+// Pass by reference
+MathToken::MathToken(TokenType t, std::wstring& s)
+{
+    this->type = t;
+    this->str = s;
+}
+
+// Pass directly
+MathToken::MathToken(TokenType t, std::wstring s)
+{
+    this->type = t;
+    this->str = s;
+}
+
+// Empty Token
+MathToken::MathToken()
+{
+    this->type = MathToken::TokenType::EOFToken;
+    this->str = L"";
+}
+
+MathTokenizer::MathTokenizer(std::wstring input)
```

Comment:
I'm still seeing this function take the input by value which is making an expensive string copy. What is the lifetime of MathTokenizer? I imagine it's short and this can just take a wstring_view or wstring by reference.